File: tests/basic.py
import websockets
import asyncio
import json


async def hello():
    loop = asyncio.get_running_loop()
    async with websockets.connect("ws://localhost:8000/ws") as ws:
        print(await ws.recv())
        ready = False
        while ws.open:
            data = json.loads(await ws.recv())
            if data["t"] == "Hello":
                print("Ready")
                await ws.send(
                    json.dumps({"t": "Stdin", "c": "uname -r && env && df -h\n"})
                )
            elif data["t"] == "Stdout" or data["t"] == "Stderr":
                print(data["c"])


asyncio.run(hello())

File: tests/basic.ts
import WebSocket from "ws";

const ws = new WebSocket("ws://localhsot:8000/ws");

ws.on('error', console.error);

ws.on('open', function open() {
  ws.send('something');
});

ws.on('message', function message(data) {
  const payload = JSON.parse(data);
  if (payload.t === "Ready") {
    ws.send(JSON.stringify({
      t: "Stdin",
      c: "pacman -Syyu --noconfirm",
    }));
  } else if (payload.t === "Stdout" || payload.t === "Stderr") {
    console.log(payload.c);
  };
});
